read-fonts/src/tables/gpos/closure.rs

```diff
@@ -17,3 +24,230 @@ impl Gpos<'_> {
         script_list.collect_features(head_ptr, &feature_list, scripts, languages, features)
     }
 }
+
+impl PositionLookupList<'_> {
+    pub fn closure_lookups(
+        &self,
+        glyph_set: &IntSet<GlyphId>,
+        lookup_indices: &mut IntSet<u16>,
+    ) -> Result<(), ReadError> {
+        let mut c = LookupClosureCtx::new(glyph_set);
+
+        let lookups = self.lookups();
+        for idx in lookup_indices.iter() {
+            let lookup = lookups.get(idx as usize)?;
+            lookup.closure_lookups(&mut c, idx)?;
+        }
+
+        lookup_indices.union(c.visited_lookups());
+        lookup_indices.subtract(c.inactive_lookups());
+        Ok(())
+    }
+}
+
+impl LookupClosure for PositionLookup<'_> {
+    fn closure_lookups(
+        &self,
+        c: &mut LookupClosureCtx,
+        lookup_index: u16,
+    ) -> Result<(), ReadError> {
+        if c.lookup_visited(lookup_index) {
+            return Ok(());
+        }
+
+        c.set_lookup_visited(lookup_index);
```

Comment:
could this be one method, like `HashSet::insert`, that returns a bool indicating if it was already visited or not?